server/scripts/budgetTest.mjs

```diff
@@ -0,0 +1,65 @@
+import mongoose from "mongoose";
+import { Budget } from "../models/Budget.mjs";
+
+async function insertBudgetIncomes() {
+    try {
+        // Connect to MongoDB
+        await mongoose.connect('mongodb+srv://marvinxia09:FKm0XWSFeaYTI8ec@cluster0.vkbcnyo.mongodb.net/', {
+            useNewUrlParser: true,
+            useUnifiedTopology: true
+        });
+
+    // Test budget data
+    const testBudget = [
+        { 
+            budgetName: "Tokyo plane ticket", 
+            goalAmount: 1000, 
+            description: "plane ticket for Japan trip",
+            goalDate: new Date("2024-06-01"),
+            status: "completed",
+        },
+        { 
+            budgetName: "Japan trip", 
+            goalAmount: 5000, 
+            description: "food, gifts, sight seeing, etc",
+            goalDate: new Date("2025-09-01"),
+            status: "in progress",
+        },
+        { 
+            budgetName: "nest egg", 
+            goalAmount: 10000, 
+            description: "funds for a rainy day",
+            goalDate: new Date("2025-06-01"),
+            status: "in progress",
+        },
+        { 
+            budgetName: "Suzuki Hayabusa", 
+            goalAmount: 10000, 
+            description: "vroom vroom",
+            goalDate: new Date("2023-05-11"),
+            status: "canceled",
+        },
+        { 
+            budgetName: "New Car", 
+            goalAmount: 10000, 
+            description: "Replace old clunker",
+            goalDate: new Date("2022-05-11"),
+            status: "completed",
+        }
+    ];
+
+    // Insert each test budget into the database
+    for (const budget of testBudget) {
+        await Budget.create(budget);
+    }
+
+    console.log("Test budget inserted successfully.");
```

Comment:
uhhhh, how do you actually know it was inserted correctly? There are no ending of the test to actually prove the point it was inserted correctly ;) AAA pattern might be great (and usually, mocking or test databases etc. if you really want to go crazy in industry)